### ISSUE TITLE
break the build if there is a Sonar violation that Sorald can fix

### ISSUE THREAD

**Issue by monperrus** [INRIA/spoon#3788]
See https://github.com/SpoonLabs/sorald/

Plus, we could have a Github PR suggestion with the Sorald fix.

**Comment by slarse** [comment]
I will have a look at this by the end of this week/start of next.

**Comment by slarse** [comment]
I'm putting this on hold until we've got a 1.1 release of Sorald. I expect that we'll have that within the next few weeks, we just need to verify that the new Spoon element/Sonar violation matching heuristic is actually doing what it should.

**Comment by slarse** [comment]
We're making progress on this! There's a new GitHub Action (https://github.com/SpoonLabs/sorald-buildbreaker) that simply runs Sorald on a project and breaks the build if it finds a repairable violation.

Before we can adopt this in Spoon, we need ratcheting (see SpoonLabs/sorald-buildbreaker#8)
